xbmc/guilib/GUITexture.cpp

```diff
@@ -365,7 +366,16 @@ bool CGUITexture::AllocResources()
     if (m_isAllocated != NORMAL)
     { // use our large image background loader
       CTextureArray texture;
-      if (CServiceBroker::GetGUI()->GetLargeTextureManager().GetImage(m_info.filename, texture, !IsAllocated(), m_use_cache))
+      if (m_requestWidth == REQUEST_SIZE_UNSET && m_requestHeight == REQUEST_SIZE_UNSET)
+      {
+        CGraphicContext& gfxContext = CServiceBroker::GetWinSystem()->GetGfxContext();
+        m_requestWidth = (int)(m_width / gfxContext.GetGUIScaleX() + 0.5f);
+        m_requestHeight = (int)(m_height / gfxContext.GetGUIScaleY() + 0.5f);
```

Comment:
One concern here is that, when being rendered, the size is the difference of the two positions that are rounded individually, so the size calculated here can differ by 1 pixel from the actual size.
I've thought of calculating the size like `(int)((m_posX + m_width) / guiScaleX) - (int)(m_posX / guiScaleX)`, but `m_posX` and `m_posY` isn't global anyway, so I'm not sure if that can really solve the problem.

Reply:
I'd have to look at the element you have issues with, but I can think of scenarios where this might be the case.

Our skinning system operates on a fixed grid, defined by the skin. If the grid size differs to the rendering resolution (let's say, 1280x720 grid size to 1920x1080 screen), our skinning system "snaps" vertex coordinates to their next position on the screen grid. It is not unusual to see such an element having different dimensions depending on its position. I guess this is baggage from way back, where most of the graphics pipeline was fixed function/software driven.

I've started to modernize the font geometry handling (on GL/GLES), which had similar issues (https://github.com/xbmc/xbmc/pull/24605). Somewhere I do have a branch doing the same for textures, but it is not ready yet.

If I were you, I would only test 1080p skins at 1920x1080 or 4k, and ignore other resolutions for now.

Reply:
I've tested this at 4K and it's working as expected.